utilities/cli/container.py

```
@@ -726,6 +726,27 @@ def get_gpu_runtime_args(self) -> List[str]:
         return args
 
+    def is_valid_endpoint(self) -> bool:
+        """Check if SCCACHE_MEMCACHED_ENDPOINT is valid"""
+        endpoint = os.environ.get("SCCACHE_MEMCACHED_ENDPOINT")
+
+        if endpoint:
+            try:
+                host, port_str = endpoint.rsplit(":", 1)
+                port = int(port_str)
+
+                with socket.create_connection((host, port), timeout=5):
```

Ruff (0.15.0)

[error] Undefined name `socket`

(F821)